pkg/api/v1beta3/types.go

```diff
@@ -1641,3 +1641,34 @@ type SecretList struct {
 
 	Items []Secret `json:"items" description:"items is a list of secret objects"`
 }
+
+// Type and constants for component health validation.
+type ComponentConditionType string
+
+// These are the valid conditions for the component.
+const (
+	ComponentHealthy ComponentConditionType = "Healthy"
+)
+
+type ComponentCondition struct {
+	Type    ComponentConditionType `json:"type"`
+	Status  ConditionStatus        `json:"status"`
+	Message string                 `json:"message,omitempty" description:"health check message received from the component"`
+	Error   string                 `json:"error,omitempty" description:"error code from health check attempt (if any)"`
+}
+
+// ComponentStatus (and ComponentStatusList) holds the cluster validation info.
+type ComponentStatus struct {
```

Comment:
Given where we're headed, I think we're going to need a full-blown Component object, with Spec as well as Status -- the same pattern as Pod and the other objects. The /components API would act as the component registry.

If we assume we're going to run all the components in pods, then we can use existing Pod mechanisms to perform liveness and readiness probes, but there's still the issue of how to collect the results of the probes. 

Potentially we'd also define services (possibly headless) for components, especially if replicated. We've discussed copying readiness condition status into Endpoints. We could maybe directly use that info, or mimic the pattern and copy the conditions from selected pods into the ComponentStatus.

We could go ahead with what you've done, but I'm concerned it won't actually last very long. If we did go ahead with this, I'd consider ComponentStatus a "virtual resource" -- it's not actually persisted in etcd. Therefore, we probably could use ListMeta in place of ObjectMeta, in which case the custom Name field could remain.

It's probably worth a VC to discuss before you do more work on this.